python/revng/scripting/daemon_project.py

```diff
@@ -0,0 +1,201 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import json
+from typing import Any, List
+
+import yaml
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+from revng.model import Binary, dump_diff_to_string, load_model_from_string, make_diff
+
+from .project import Project
+
+
+class DaemonProject(Project):
+    def __init__(self, url: str):
+        self.transport = AIOHTTPTransport(url=f"{url}/graphql/")
+        self.client = Client(
+            transport=self.transport, fetch_schema_from_transport=True, execute_timeout=None
+        )
+        super().__init__()
+
+    def import_binary(self, path: str) -> None:
+        query = gql(
+            """
+            mutation upload($file: Upload!) {
+                uploadFile(file: $file, container: "input")
+            }
+        """
+        )
+
+        with open(path, "rb") as binary_file:
+            result = self.client.execute(
+                query, variable_values={"file": binary_file}, upload_files=True
+            )
+
+        if not result["uploadFile"]:
+            raise RuntimeError("File upload failed")
+
+    def import_and_analyze(self, input_binary: str) -> None:
+        self.import_binary(input_binary)
+        self._analyses_list("revng-initial-auto-analysis")
+        self._analyses_list("revng-c-initial-auto-analysis")
+
+        self.pipeline_description = yaml.safe_load(self._get_pipeline_description())
+        self.set_model(self._get_model())
```

Comment:
This function should probably be moved to `project.py`, the functionality is the same, modulo not using `self._analyses_list`